### PR TITLE
[py] improve driver logging

### PR DESCRIPTION
**Pull request by titusfortner**
### Description
* Deprecate the current `log_file` or `log_path` Service class constructor parameters, which only accepts strings as filenames
* Create new `log_output` (which matches Java's `withLogOutput()`) constructor parameter. This parameter that accepts:
    * String for name of file to create (so just like the current methods)
    * TextIOWrapper as an already open file (addresses #10703)
    * int (for explicitly specifying `subprocess.STDOUT` or `subprocess.DEVNULL`)
* Default for Firefox is now null output instead of `geckodriver.log` (#12016)
* If log file selected, Chromium and IE now pipe the whole process output to a file rather than using driver argument (this matches what Java and Ruby do, and captures all driver related output which is nice)
    * EXCEPT - if `--append-log` or `--readable-timestamp` are passed (chromium) because those only work with `log-file` driver argument
* Safari - quiet parameter does not do anything in Apple's driver; it was never removed from prior implementation 

### Motivation and Context
Make things consistent between drivers and with Java & Ruby

One thing I'm not sure of....
This currently deprecates and warns about the change in default Firefox behavior; I think I'd prefer just changing it and maybe logging a message about the change with a link to the docs (https://www.selenium.dev/documentation/webdriver/drivers/service/#setting-log-output)

Also, I'm not sure I have the typing exactly right? I'm not sure what `typing.IO[typing.Any]` actually includes.

### CODE REVIEW

**Review by diemol — APPROVED** [review]
Thank you, @titusfortner!

**Comment by vringar** [comment]
Thanks for implementing this!
As this got merged in the 4.11 merge window, will it get released on July 21? And are there any beta/preview/nightly artifacts published somewhere to use this immediately?

**Comment by titusfortner** [comment]
@vringar take a look at #12207

File: py/selenium/webdriver/common/service.py [hunk 1]
```diff
@@ -51,14 +52,27 @@ def __init__(
...
         env: typing.Optional[typing.Mapping[typing.Any, typing.Any]] = None,
         start_error_message: typing.Optional[str] = None,
         **kwargs,
     ) -> None:
+        if isinstance(log_output, str):
+            self.log_output = open(log_output, "a+", encoding="utf-8")
+        elif log_output is subprocess.STDOUT:
+            self.log_output = None
+        elif log_output is None or log_output is subprocess.DEVNULL:
+            self.log_output = open(os.devnull, "wb")
```
Comment (asottile-sentry) [inline py/selenium/webdriver/common/service.py:66]: these file descriptors are never closed leading to ResourceWarnings when this is garbage collected (and leaked fds)

**Pull request merged**